packages/lib-utils/src/k8s/k8s-errors.ts

```diff
@@ -0,0 +1,11 @@
+import { CustomError } from '@openshift/dynamic-plugin-sdk';
+import type { K8sStatus } from '../types/k8s';
+
+/**
+ * Error class used when Kubernetes cannot handle a request.
+ */
+export class K8sStatusError extends CustomError {
+  constructor(readonly statusObject: K8sStatus) {
```

Comment:
Nit: I'd omit the "Object" suffix for simplicity, since `K8sStatus` refers to object type definition anyway.

```suggestion
  constructor(readonly status: K8sStatus) {
```